pkg/dataServer/dataServer.go

```diff
@@ -43,6 +44,11 @@ func CreateServer(
 	if err != nil {
 		return nil, errors.Wrapf(err, "creating data server tracker runner instance")
 	}
+
+	filterLogger, err := util.ApplyFilter(*config, "DataServer", logger)
```

Comment:
No need to choose a differnt name and lets use the idea for `const ComponentName = "DataServer"`
```suggestion
	logger, err := util.ApplyFilter(*config, ComponentName, logger)
```

later in the code it can be like
```

logger:       log.With(filterLogger, "component", ComponentName)}, nil
```